inst/templates/ad_advs.R

```diff
@@ -0,0 +1,377 @@
+  ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 1,
+  "HEIGHT", "HEIGHT", "Height (cm)", 2,
+  "BMI", "BMI", "Body Mass Index(kg/m^2)", 3,
+  "HDCIRC", "HDCIRC", "Head Circumference (cm)", 4
+)
+attr(param_lookup$VSTESTCD, "label") <- "Vital Signs Test Short Name"
+
+# Get list of DM vars required for derivations
+dm_vars <- exprs(SEX, BRTHDTC)
+advs <- vs %>%
+  # Join DM with VS (need BRTHDT for AAGECUR derivation)
+  derive_vars_merged(
+    dataset_add = dm,
+    new_vars = dm_vars,
+    by_vars = exprs(STUDYID, USUBJID)
+  ) %>%
+  mutate(BRTHDT = convert_dtc_to_dt(
+    BRTHDTC,
+    highest_imputation = "n",
+    date_imputation = "first",
+    min_dates = NULL,
+    max_dates = NULL,
+    preserve = FALSE
+  )) %>%
+  ## Calculate ADT ----
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = VSDTC
+  ) %>%
+  ## Calculate Current Analysis Age AAGECUR ----
+  derive_vars_aage(
+    start_date = BRTHDT,
+    end_date = ADT,
+    age_unit = "DAYS",
+    type = "interval"
+  ) %>%
+  rename(
+    AAGECUR = AAGE,
+    AAGECURU = AAGEU
+  )
+
+advs <- advs %>%
+  ## Add PARAMCD only - add PARAM etc later ----
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAMCD),
+    by_vars = exprs(VSTESTCD)
+  ) %>%
+  ## Calculate AVAL and AVALC ----
+  mutate(
+    AVAL = VSSTRESN,
+    AVALC = VSSTRESC
+  )
+
+## Get visit info ----
+# See also the "Visit and Period Variables" vignette
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html#visits)
+windows <- tribble(
+  ~AVISIT,     ~AVISITN,
+  "Screening",        0,
+  "Day 1",            1,
+  "6 Months",         2,
+  "12 Months",        3
+)
+
+advs <- advs %>%
+  # Derive Timing
+  mutate(
+    AVISIT = case_when(
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      !is.na(VISIT) ~ str_to_title(VISIT),
+      TRUE ~ NA_character_
+    )
+  ) %>%
+  derive_vars_merged(
+    dataset_add = windows,
+    by_vars = exprs(AVISIT)
+  )
+
+## Add PARAM/PARAMN ----
+advs <- advs %>%
+  # Derive PARAM and PARAMN
+  derive_vars_merged(dataset_add = select(param_lookup, -VSTESTCD), by_vars = exprs(PARAMCD))
+
+## Merge ADVS to the chosen Growth metadata as an input to meta_criteria ----
+## Calculate z-scores/percentiles
+## Calculate Weight for AGE z-score and Percentile ----
+## Note: PARAMN needs to be updated.
+advs_wgt_age <- derive_params_growth_age(
+  advs,
+  sex = SEX,
+  age = AAGECUR,
+  age_unit = AAGECURU,
+  meta_criteria = weight_for_age,
+  parameter = VSTESTCD == "WEIGHT",
+  set_values_to_sds = exprs(
+    PARAMCD = "WTASDS",
+    PARAM = "Weight-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "WTAPCTL",
+    PARAM = "Weight-for-age percentile"
+  )
+) %>%
+  mutate(
+    PARAMN = case_when(
+      PARAMCD == "WTASDS" ~ 5,
+      PARAMCD == "WTAPCTL" ~ 6,
+      TRUE ~ PARAMN
+    )
+  ) %>%
+  filter(PARAMCD %in% c("WTASDS", "WTAPCTL"))
```

Comment:
i dont think needed as fine to keep source params too

Reply:
See from line 105 in https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_advs.R - we usually just build up the ADaM by taking VS records and then adding one new derived parameter each time by piping to the next `derive_param_xxx()` function. Our `parameter = VSTESTCD == "WEIGHT"` argument for example would choose which input records to use for the derived parameter each time. Does that make sense?